content/specs/auth.md

```diff
@@ -0,0 +1,401 @@
+
+Authorization Servers should separately implement a web interface which allows authenticated users to view active OAuth sessions and delete them.
+
+### Summary of Authorization Flow
+
+This is a high-level description of what an atproto OAuth authorization flow looks like. It assumes the user already has an atproto account.
+
+The client starts by asking for the user’s account identifier (handle or DID), or for a PDS/entryway hostname. See “Identity Authentication” section for details.
+
+For an account identifier, the client resolves the identity to a DID document, extracts the declared PDS URL, then fetches the Resource Server and Authorization Server locations. If starting with a server hostname, the client resolves that hostname to an Authorization Server. In either case, Authorization Server metadata is fetched and verified against requirements for atproto OAuth (see “Authorization Server” section).
+
+The client next makes a Pushed Authorization Request via HTTP POST request. See “Authorization Request” section; some notable details include:
+
+- a randomly generated `state` token is required, and will be used to reference this authorization request with the subsequent response callback
+- PKCE is required, so a secret value is generated and stored, and a derived challenge is included in the request
+- `scope` values are requested here, and must include `atproto`
+- for confidential clients, a `client_assertion` is included, with type `jwt-bearer`, signed using the secret client authentication key
+- the client generates a new DPoP key for the user/session and uses it starting with the PAR request
+- if the auth flow started with an account identifier, the client should pass that starting identifier via the `login_hint` field
+- atproto uses PAR, so the request will be sent as an HTTP POST request to the Authorization Server
+
+The Authorization Server will receive the PAR request and use the `client_id` URL to resolve the client metadata document. The server validates the request and client metadata, then stores information about the session, including binding a DPoP key to the session. The server returns a `request_uri` token to the client, including a DPoP nonce via HTTP header.
+
+The client receives the `request_uri` and prepares to redirect the user. At this point, the client usually needs to persist information about the session to some type of secure storage, so it can be read back after the redirect returns. This might be a database (for a web service backend) or web platform storage like IndexedDB (for a browser app). The client then redirects the user via browser to the Authorization Server’s auth endpoint, including the `request_uri` as a URL parameter.
```

Comment:
This seems like a good place to remind of:

```suggestion
The client receives the `request_uri` and prepares to redirect the user. At this point, the client usually needs to persist information about the session to some type of secure storage, so it can be read back after the redirect returns. This might be a database (for a web service backend) or web platform storage like IndexedDB (for a browser app). The client then redirects the user via browser to the Authorization Server’s auth endpoint, including the `request_uri` as a URL parameter. In any case, clients must not store session data directly in the `state` request param.
```